### ISSUE TITLE
Stack error log on 1.18.0 release

### ISSUE THREAD

**Issue by acheroncrypto** [solana-labs/solana#35003]
### Problem

Building programs with `cargo-build-sbf` results in the following error log:

```
Error: Function _ZN112_$LT$solana_program..instruction..InstructionError$u20$as$u20$solana_frozen_abi..abi_example..AbiEnumVisitor$GT$13visit_for_abi17ha3a75fe06f2a2af4E Stack offset of 4584 exceeded max offset of 4096 by 488 bytes, please minimize large stack variables
```

The compilation still works but the error message can be confusing to people.

I'm able to reproduce this issue from any program while using `1.18.0` version for both `solana-program` and `solana-cli`.

**Comment by jedusor1** [comment]
Error: Function _ZN112_$LT$solana_program..instruction..InstructionError$u20$as$u20$solana_frozen_abi..abi_example..AbiEnumVisitor$GT$13visit_for_abi17h45201742ade77720E Stack offset of 4584 exceeded max offset of 4096 by 488 bytes, please minimize large stack variables

Same here, can confirm this is still and issue. Then the solana program build step after doesn't seem to see the file being referenced after that....very confusing

**Comment by joncinque** [comment]
Please upgrade to a newer version of `solana-program`, the fix is part of 1.18.2 and above
